### PR TITLE
feat: add location selection on full screen map

### PR DESCRIPTION
**Pull request by ardasnturk**
- turned `onMapPress` into a separate method to send it as param to `MapViewScreen`
- added `position` state to `MapViewScreen` to add pins to the map
- updated the `locationsWithPin` state in the `useEffect` to prevent two pins from being added when clicking on the map in `MapViewScreen`
- added `onMyLocationButtonPress` as an extra method to use the current location information on the full screen map
- added `updateRegion` to refresh the map when the user presses the current location button

SUE-80

## Screenshots:

|||
|--|--|
Simulator Screenshot - iPhone 15 Pro Max - 2024-06-12 at 20 50 13|image

### CODE REVIEW

File: src/components/SUE/report/SueReportLocation.tsx [hunk 1]
```diff
@@ -187,53 +252,20 @@ export const SueReportLocation = ({
...
-
-              try {
-                await handleGeocode(nativeEvent.coordinate);
-              } catch (error) {
-                setSelectedPosition(undefined);
-                Alert.alert(texts.sue.report.alerts.hint, error.message);
-              }
-            }
-          }}
+          onMyLocationButtonPress={() => onMyLocationButtonPress()}
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:255]: could be `onMyLocationButtonPress={onMyLocationButtonPress}` because there are not params passed, right?

File: src/screens/MapViewScreen.js [hunk 2]
```diff
@@ -38,11 +37,10 @@ export const MapViewScreen = ({ navigation, route }) => {
     <>
       <Map
         {...{
-          calloutTextEnabled,
           geometryTourData,
           isMaximizeButtonVisible,
           locations,
-          mapStyle: styles.map,
+          mapStyle: styles.mapStyle,
```
Comment (donni106) [inline src/screens/MapViewScreen.js:43]: 🖐️

File: src/config/navigation/defaultStackConfig.tsx [hunk 3]
```diff
@@ -374,6 +375,11 @@ export const defaultStackConfig = ({
         usedAsInitialScreen: true
       }
     },
+    {
+      routeName: ScreenName.SueReportMap,
```
Comment (donni106) [inline src/config/navigation/defaultStackConfig.tsx:379]: `SueReportMapView`?

File: src/types/Navigation.ts [hunk 4]
```diff
@@ -55,6 +55,7 @@ export enum ScreenName {
   SueList = 'SueList',
   SueMap = 'SueMap',
   SueReport = 'SueReport',
+  SueReportMap = 'SueReportMap',
```
Comment (donni106) [inline src/types/Navigation.ts:58]: `SueReportMapView`?

Comment (ardasnturk) [reply src/types/Navigation.ts]: https://github.com/smart-village-solutions/smart-village-app-app/commit/5142afcd38040ba6d3c8df434c98880a3effb78e

File: src/screens/SUE/SueMapViewScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import { StackNavigationProp } from '@react-navigation/stack';
+import React, { useContext, useEffect, useState } from 'react';
+import { StyleSheet } from 'react-native';
+
+import { SettingsContext } from '../../SettingsProvider';
+import { Map } from '../../components';
+import { colors, normalize } from '../../config';
+
+export const SueMapViewScreen = ({
+  navigation,
```
Comment (donni106) [inline src/screens/SUE/SueMapViewScreen.tsx:10]: not used

File: src/screens/SUE/SueMapViewScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    mapCenterPosition,
+    onMapPress,
+    onMarkerPress,
+    onMyLocationButtonPress,
+    selectedPosition,
+    showsUserLocation
+  } = route?.params ?? {};
+
+  const { globalSettings } = useContext(SettingsContext);
+  const { navigation: navigationType } = globalSettings;
```
Comment (donni106) [inline src/screens/SUE/SueMapViewScreen.tsx:32]: not used

File: src/screens/SUE/SueMapViewScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  }
+});
+
+const stylesWithProps = ({ navigationType }) => {
+  return StyleSheet.create({
+    position: {
+      bottom: navigationType === 'drawer' ? '8%' : '4%'
+    }
+  });
+};
```
Comment (donni106) [inline src/screens/SUE/SueMapViewScreen.tsx:131]: not used

File: src/components/SUE/report/SueReportLocation.tsx [hunk 8]
```diff
@@ -177,6 +177,71 @@ export const SueReportLocation = ({
...
+              } catch (error) {
+                setSelectedPosition(undefined);
+                Alert.alert(texts.sue.report.alerts.hint, error.message);
+              }
+            }
+          }
+        }
+      ]);
+    } else {
+      const location = position || lastKnownPosition;
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:227]: why ist there no alert here for the full map?

File: src/components/SUE/report/SueReportLocation.tsx [hunk 9]
```diff
@@ -177,6 +177,71 @@ export const SueReportLocation = ({
...
+        setUpdatedRegion(true);
+        setUpdateRegionFromImage(false);
+        setSelectedPosition(location.coords);
+
+        try {
+          await handleGeocode(location.coords);
+        } catch (error) {
+          setSelectedPosition(undefined);
+          Alert.alert(texts.sue.report.alerts.hint, error.message);
+          throw new Error(error.message);
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:239]: why is this throw here but not in the part for the small map?

File: src/components/SUE/report/SueReportLocation.tsx [hunk 10]
```diff
@@ -187,53 +252,19 @@ export const SueReportLocation = ({
...
-
-              try {
-                await handleGeocode(nativeEvent.coordinate);
-              } catch (error) {
-                setSelectedPosition(undefined);
-                Alert.alert(texts.sue.report.alerts.hint, error.message);
-              }
-            }
-          }}
+          onMyLocationButtonPress={() => onMyLocationButtonPress()}
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:255]: `onMyLocationButtonPress={onMyLocationButtonPress}`?

File: src/screens/MapViewScreen.js [hunk 11]
```diff
@@ -78,7 +76,7 @@ const styles = StyleSheet.create({
   augmentedRealityInfoContainer: {
     width: '90%'
   },
-  map: {
+  mapStyle: {
```
Comment (donni106) [inline src/screens/MapViewScreen.js:79]: no! why changing it? now it is different to the new sue map view screen. this is not related to any topic of the PR. please keep the file as it was.

File: src/components/SUE/report/SueReportLocation.tsx [hunk 12]
```diff
@@ -177,6 +177,71 @@ export const SueReportLocation = ({
...
+  const onMyLocationButtonPress = async (isFullScreenMap = false) => {
+    if (!isFullScreenMap) {
+      Alert.alert(texts.sue.report.alerts.hint, texts.sue.report.alerts.myLocation, [
+        {
+          text: texts.sue.report.alerts.no
+        },
+        {
+          text: texts.sue.report.alerts.yes,
+          onPress: async () => {
+            const location = position || lastKnownPosition;
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:209]: couldn't this happen earlier to get the `currentPosition` (even better wording than `location` i guess)? now it happens twice in the method here and again a third time at line 260.

File: src/screens/SUE/SueMapViewScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    onMarkerPress,
+    onMyLocationButtonPress,
+    selectedPosition,
+    showsUserLocation
+  } = route?.params ?? {};
+
+  const { globalSettings } = useContext(SettingsContext);
+  const { navigation: navigationType } = globalSettings;
+
+  const [position, setPosition] = useState(selectedPosition);
```
Comment (donni106) [inline src/screens/SUE/SueMapViewScreen.tsx:34]: i was confused that this is named `setPosition` instead of `setSelectedPosition` like in the other file. it does the same, or am i wrong?

File: src/components/SUE/report/SueReportLocation.tsx (hunk 8 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx:227]: This is because `SueMapViewScreen` is used to set the position within itself to set the position.

File: src/components/SUE/report/SueReportLocation.tsx (hunk 9 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx:239]: Because I want to be able to recognise an error in `SueMapViewScreen` and make `setPosition` `undefined`. No need to create errors in the small map view. Because the `setSelectedPosition` value is updated as `undefined` in the catch section.

File: src/screens/MapViewScreen.js (hunk 11 above)
Comment (ardasnturk) [reply src/screens/MapViewScreen.js:79]: I need to update this to `mapStyle`. Otherwise it will not match `styles.mapStyle` in line 43. Or I need to update the style in line 43 to `styles.map`.

File: src/components/SUE/report/SueReportLocation.tsx (hunk 12 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx:209]: Yes, that's right. I'll create this at the top of the `onMyLocationButtonPress`. 👍 I will update it as `currentPosition`.

File: src/screens/SUE/SueMapViewScreen.tsx (hunk 13 above)
Comment (ardasnturk) [reply src/screens/SUE/SueMapViewScreen.tsx:34]: That's right. Actually, they both have the same function. I only used this nomenclature because we have the `selectedPosition` that comes with the prop. but I'll update the prop's name to `position`.

File: src/screens/SUE/SueMapViewScreen.tsx (hunk 5 above)
Comment (ardasnturk) [reply src/screens/SUE/SueMapViewScreen.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1140/commits/e01b9da4ca5a57b77cbd45cf023113332406e9a6

File: src/screens/MapViewScreen.js (hunk 11 above)
Comment (ardasnturk) [reply src/screens/MapViewScreen.js]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1140/commits/121e12118e6a32cde20cd6db6267c2a568c3d217

File: src/components/SUE/report/SueReportLocation.tsx [hunk 14]
```diff
@@ -187,53 +250,19 @@ export const SueReportLocation = ({
...
-            }
-          }}
+          onMyLocationButtonPress={onMyLocationButtonPress}
+          onMapPress={onMapPress}
           onMaximizeButtonPress={() =>
-            navigation.navigate(ScreenName.MapView, {
+            navigation.navigate(ScreenName.SueReportMapView, {
               calloutTextEnabled: true,
-              locations
+              currentPosition: position || lastKnownPosition,
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:258]: we could use `currentPosition` here as well, if not defined inside the method but right in the component, right?

File: src/components/SUE/report/SueReportLocation.tsx [hunk 15]
```diff
@@ -177,6 +177,69 @@ export const SueReportLocation = ({
...
 
+  const onMapPress = async ({
+    nativeEvent
+  }: {
+    nativeEvent: { action: string; coordinate: Location.LocationObjectCoords };
+  }) => {
+    if (nativeEvent.action !== 'marker-press' && nativeEvent.action !== 'callout-inside-press') {
+      setUpdatedRegion(false);
+      setUpdateRegionFromImage(false);
+      setSelectedPosition(nativeEvent.coordinate);
```
Comment (donni106) [inline src/components/SUE/report/SueReportLocation.tsx:188]: we could call this always first where used, maybe this helps a bit for "speed of visual feedback" when pressing the map? it shouldn't hurt to move it up, right?

File: src/components/SUE/report/SueReportLocation.tsx (hunk 14 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx:258]: Yes 🙌

File: src/components/SUE/report/SueReportLocation.tsx (hunk 15 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx:188]: Are you talking about `setSelectedPosition`?

File: src/components/SUE/report/SueReportLocation.tsx (hunk 14 above)
Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1140/commits/2cb8a45bc25d8174f8fa77cde39f44c1f838be48

File: src/components/SUE/report/SueReportLocation.tsx (hunk 15 above)
Comment (donni106) [reply src/components/SUE/report/SueReportLocation.tsx:190]: yes

Comment (ardasnturk) [reply src/components/SUE/report/SueReportLocation.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1140/commits/1f80a77877fa798a7f4fc8c4d88ef7cbd3af53ef

**Review by donni106 — APPROVED** [review]
